sdk/src/main/java/io/opentelemetry/sdk/trace/SpanInterceptor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.trace.Span;
+
+/**
+ * SpanInterceptor is the interface {@code TracerSdk} uses to allow synchronous hooks for when a
+ * {@code Span} is started or when a {@code Span} is ended.
+ */
+public interface SpanInterceptor {
+  /**
+   * Called when a {@link io.opentelemetry.trace.Span} is started, if the {@link
+   * Span#isRecordingEvents()} returns true.
+   *
+   * @param span the {@code SpanSdk} that just started.
+   */
+  void onStartSync(SpanSdk span);
```

Comment:
at this stage SpanContext is not modifiable, correct? One may need to update tracestate with the span's or resource's information here.

Reply:
Maybe `onBeforeSpanCreated` of sort?

Reply:
This is called immediately after the Span is started. To modify the Tracestate we need a different hook "onBuildSpan" which will pass individual items (a Span is not created yet).

Reply:
so the scenario of onBuildSpan will not be handled by interceptors?

Reply:
I don't need it right now, and I am not sure about how the API should look like. We can have a SpanBuilderInterceptor for that?